### PR TITLE
Justify nginx proxy settings to make port mappings support using the host port other than 80.

### PR DESCRIPTION
**Pull request by myme5261314**
Refer to #15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,6 +103,7 @@ If the environment variable "SUPERVISED" is set to true, the container won't be
 
 ```conf
 server.port_mappings = 80:80,443:443
+server.port_mappings = XXXX:80,XXXX:443
 ```
```
Comment (odelrio) [inline README.md]: Please, explain that XXXX has to be replaced with the host (local) port.

**Comment by freeplant** [comment]
Please try the latest version. `launcher` is no longer used.

**Comment by noonowl71** [comment]
@freeplant it's not working with latest version. This pull request is still needed.

With latest version if you map to another port through docker nginx redirects to port 80:

image

**Comment by noonowl71** [comment]
Changing `proxy_set_header Host $host;` with `proxy_set_header Host $http_host;` does the job:

image

File: README.md [hunk 2]
```diff
@@ -103,6 +103,7 @@ If the environment variable "SUPERVISED" is set to true, the container won't be
 
 ```conf
 server.port_mappings = 80:80,443:443
+server.port_mappings = XXXX:80,XXXX:443
```
Comment (shoeper) [inline README.md]: I would remove this change and add a hint that is possible to configure another port.

**Comment by noonowl71** [comment]
I think this PR is not about changing default nginx port. Current port is
fine, the problem is when you expose the port to a different port on
docker. It's not working because it keeps redirecting to 80. The root of
the problem is that nginx $host variable does not include the port, so the
header is set without the port. $http_host includes both domain and port
and works ok.

**Comment by BeaverCC** [comment]
Will this PR be merged or closed？
Delete this line 
```
proxy_set_header Host $host;
```
from the template file is ok for me.

**Comment by shoeper** [comment]
The issue I have is that http_host is set by the client. It would be better to set X-Forwarded-Port, but Django does not support it out of the box.

**Comment by freeplant** [comment]
Simply changing to `proxy_set_header Host $http_host;` will cause problem when you automatically get let's encrypt certification. Because The let's encrypt will visit 80 port.

We want keep our docker image simple. So advanced usage case like using another port (not 80) will not be supported in docker way. Otherwise, we will force the majority users to manually set up let's encrypt certification.

**Pull request closed without merging**